sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -98,8 +96,10 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
         l,
         projects,
         filters,
-        (a, f) =>
-          toCatalystRDD(l, a, t.buildScan(a.map(_.name).toArray, f, t.paths, confBroadcast))) :: Nil
+        (a, f) => {
+          t.refresh()
```

Comment:
Ah, this is really not acceptable. We can't afford a refresh for every data source table scan... I missed those two similar `refresh()` calls when merging #7696. I'm trying to figuring out a more reasonable fix for this.